frontend/src/lib/components/FlowBuilder.svelte

```diff
@@ -427,6 +428,15 @@
 		loadingSave = true
 		try {
 			const flow = cleanInputs(flowStore.val)
+			
+			if (flow.value?.modules) {
+				for (const module of flow.value.modules) {
```

Comment:
that only validate the first level, use dfs function to go through all the modules